packages/astro/src/core/sync/index.ts

```diff
@@ -74,6 +75,16 @@ export async function sync(
 		)
 	);
 
+	// Patch `ws.send` to bubble up error events
+	// `ws.on('error')` does not fire for some reason
+	const wsSend = tempViteServer.ws.send;
+	tempViteServer.ws.send = (payload: HMRPayload) => {
+		if (payload.type === 'error') {
+			throw payload.err;
+		}
+		return wsSend(payload);
+	};
```

Comment:
I think it's because the Vite server `logLevel` is set to silent?

https://github.com/withastro/astro/blob/4516d7b22c5979cde4537f196b53ae2826ba9561/packages/astro/src/core/sync/index.ts#L71

Reply:
I was looking at [this Vite code](https://github.com/vitejs/vite/blob/d09bbd093a4b893e78f0bbff5b17c7cf7821f403/packages/vite/src/node/server/middlewares/error.ts#L43-L58) where I found the `logLevel`. Not a big deal since we're pushing to get this over the line, and we can follow up later 👍